iml-manager-cli/src/nginx.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::error::ImlManagerCliError;
+use console::Term;
+use lazy_static::*;
+use regex::{Captures, Regex};
+use std::{collections::HashMap, env, str};
+use structopt::StructOpt;
+use tokio::fs;
+
+#[derive(Debug, StructOpt)]
+pub enum NginxCommand {
+    /// Generate Nginx conf
+    #[structopt(name = "generate-config")]
+    GenerateConfig {
+        /// Set the nginx config path
+        #[structopt(short = "p", long = "path")]
+        template_path: String,
+
+        #[structopt(short = "o", long = "output")]
+        output_path: Option<String>,
+    },
+}
+
+fn replace_template_variables(contents: &str, vars: HashMap<String, String>) -> String {
+    lazy_static! {
+        static ref RE: Regex = Regex::new(r"\{\{(\w+)\}\}").unwrap();
+    }
+
+    let config: String = contents
+        .lines()
+        .map(|l| {
+            RE.replace_all(l, |caps: &Captures| {
+                let key = &caps[1];
+                let val = vars
+                    .get(key)
+                    .unwrap_or_else(|| panic!("{} variable not set", key));
+
+                caps[0].replace(&format!("{{{{{}}}}}", key), &val)
```

Comment:
Could you expand the regex to also capture the braces? If you did you can then you could probably omit this and replace directly.

Reply:
sure, sounds good.